.github/workflows/cmake.yml

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2024 Sebastian Pipping <sebastian@pipping.org>
+# Licensed under the 3-Clause BSD License
+
+name: Build using CMake
+
+# Drop permissions to minimum for security
+permissions:
+  contents: read
+
+on:
+  pull_request:
+  push:
+  schedule:
+    - cron: '0 3 * * 5'  # Every Friday at 3am
+  workflow_dispatch:
+
+jobs:
+  build_cmake:
+    name: CMake (${{ matrix.runs-on }}, ${{ matrix.cc }})
+    runs-on: ${{ matrix.runs-on }}
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          # FUSE 3 on Linux
+          - cc: gcc-13
+            clang_major_version: null
+            runs-on: ubuntu-22.04
+          - cc: clang-18
+            clang_major_version: 18
+            runs-on: ubuntu-22.04
+          # macFUSE on macOS
+          - cc: gcc-13
+            clang_major_version: null
+            runs-on: macos-12
+          - cc: clang-15
+            clang_major_version: 15
+            runs-on: macos-14
+
+    steps:
+      - name: Add Clang/LLVM repositories
+        if: "${{ runner.os == 'Linux' && contains(matrix.cc, 'clang') }}"
+        run: |-
+          set -x
+          source /etc/os-release
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+          sudo add-apt-repository "deb http://apt.llvm.org/${UBUNTU_CODENAME}/ llvm-toolchain-${UBUNTU_CODENAME}-${{ matrix.clang_major_version }} main"
+
+      - name: Install build dependencies
+        if: "${{ runner.os == 'Linux' }}"
+        run: |-
+          sudo apt-get update
+          sudo apt-get install --yes --no-install-recommends \
+            cmake \
+            libfuse3-dev \
+            pkg-config
+
+      - name: Install build dependency Clang ${{ matrix.clang_major_version }}
+        if: "${{ runner.os == 'Linux' && contains(matrix.cc, 'clang') }}"
+        run: |-
+          sudo apt-get install --yes --no-install-recommends -V \
+              clang-${{ matrix.clang_major_version }} \
+              libclang-rt-${{ matrix.clang_major_version }}-dev \
+              llvm-${{ matrix.clang_major_version }}
+
+      - name: Add versioned aliases for Clang ${{ matrix.clang_major_version }}
+        if: "${{ runner.os == 'macOS' && contains(matrix.cc, 'clang') }}"
+        run: |-
+          set -x
+          sudo ln -s "$(brew --prefix llvm@${{ matrix.clang_major_version }})"/bin/clang /usr/local/bin/clang-${{ matrix.clang_major_version }}
+
+      - name: Install build dependencies
+        if: "${{ runner.os == 'macOS' }}"
+        run: |-
+          set -x
+          brew install --cask macfuse
+
+      - name: Checkout Git branch
+        uses: actions/checkout@v4
```

Comment:
I don't know where to get the current hash, but in libfuse we got the advise to use hash-pinned workflow actions. Like this
```
bschubert2@imesrv6 libfuse.git>git show 0d7f43f110ec2a246bda906d4190767f8626446c
commit 0d7f43f110ec2a246bda906d4190767f8626446c
Author: Pedro Kaj Kjellerup Nacht <pnacht@google.com>
Date:   Wed Jul 26 13:48:07 2023 +0000

    Hash-pin workflow Actions
    
    Signed-off-by: Pedro Kaj Kjellerup Nacht <pnacht@google.com>

diff --git a/.github/workflows/codespell.yml b/.github/workflows/codespell.yml
index 1ae78d412073..233036851f85 100644
--- a/.github/workflows/codespell.yml
+++ b/.github/workflows/codespell.yml
@@ -17,6 +17,6 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
       - name: Codespell
-        uses: codespell-project/actions-codespell@v1
+        uses: codespell-project/actions-codespell@94259cd8be02ad2903ba34a22d9c13de21a74461 # v2.0
```

(Well we could use those of libfuse, but I'm sure there is a more official way).

https://github.com/libfuse/libfuse/issues/819

Reply:
@bsbernd I'm 100% with you on that but my experience is that most upstreams don't like it because of the additional noise it creates from bumping pull requests. If @rpodgorny is on board with more pull request for more security, I can make the switch in a second. PS: The current hash can be retrieve through either `git show <tag>` or on the details page of the tag or GitHub release.

Reply:
@bsbernd @rpodgorny PS: I have added a commit migrating to SHA1s and hashes now. Your call, I just need to know to keep or drop it.